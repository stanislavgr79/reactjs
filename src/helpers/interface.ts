export interface IMovie {
  [x: string]: any;
  id?: number;
  title: string;
  tagline?: string;
  vote_average?: number;
  vote_count?: number;
  release_date: string;
  movieUrl: string;
  genres: string[] | { label: string, value: string }[];
  poster_path?: string | undefined;
  overview: string;
  runtime: number | string | undefined;
}
