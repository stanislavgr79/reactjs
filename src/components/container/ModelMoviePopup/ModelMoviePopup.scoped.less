@import 'react-datepicker/dist/react-datepicker.css';

.site_name {
    padding-top: 10px;
    margin: unset;
}

.edit_movie, .add_movie {
    padding-top: 20px;
    padding-bottom: 20px;
    margin-bottom: 100%;
    display: flex;
    justify-content: center;
    
    .form {
        padding-left: 7%;
        padding-right: 7%;
        background-color: #232323;
        position: relative;
        padding-top: 50px;
        max-width: 720px;
        
        .page_title{
            font-size: 36px;
            color: #FFF;
            font-weight: 100;
        }
        .page_close{
                position: absolute;
                right: 20px;
                top: 20px;
        }
    }
}

.movie_form {
    display: flex;
    flex-wrap: wrap;
    flex-flow: column;

    & > div {
        color: #FFF;
        font-size: 16px;
        font-weight: 400;
    }

    & > Label {
        color: #F65261;
        font-weight: 600;
        margin-top: 6px;
        margin-bottom: 6px;
        font-size: 16px;
    }

    .button_section {
        display: flex;
        flex-direction: row-reverse;
        height: 100%;
        margin-top: 50px;
        margin-bottom: 40px;
    }

    input {
        background-color: #424242;
        padding: 2px 8px;
        font-size: 15px;
        border: 1px solid #424242;
        border-radius: 4px;
        color: #FFF;

        &:focus {
            border-color: blue;
        }
    }

    .simple_input {
        height: 28px;
        width: inherit;
    }

    .input_readonly{
        background-color: #232323;
        border: none;
        padding-left: 0;
    }
}

.select_genres {
    color: #bbb;

    & > div {
        background-color: #424242;
        border: none;
        font-size: 18px;

        & > div {
            color: #bbbbbb;
        }
    }
}

.select_genres [class$="control"] {
    background-color: #424242;
    border: none;
    box-shadow: none;
    border: 1px solid;
}

.select_genres label:hover {
    color: #F65261;
}

.select_genres [class$='option']{
    background-color: inherit;
    &:hover{
        background: none;
    }
    &:active {
        background-color: inherit;
        color: #F65261;
    }
}

.popup_model {
    position: fixed;
    inset: 0px;
    display: flow-root;
    z-index: 999;
    background: rgba(49, 49, 49, 0.88);
    pointer-events: auto;
  }

  .model_content {
    height: 100%;
    position: relative;
    margin: auto;
    pointer-events: auto;

    .footer{
    position: absolute;
    left: 0;
    bottom: 0;
    width: 100%;
    }
  }

  .image_calendar{
    position: absolute;
    width: 23px;
    margin-left: -28px;
    padding-top: 2px;
    &:hover{
        cursor: pointer;
    }
  }
