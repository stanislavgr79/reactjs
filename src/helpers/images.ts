const images = [
  {
    name: 'Ricky and Morty.jpg',
    value: './resources/images/Ricky and Morty.jpg',
  },
  {
    name: 'Pulp Fiction.jpg',
    value: './resources/images/Pulp Fiction.jpg',
  },
  {
    name: 'Kill Bill.jpg',
    value: './resources/images/Kill Bill.jpg',
  },
  {
    name: 'Avengers.jpg',
    value: './resources/images/Avengers.jpg',
  },
  {
    name: 'Inception.jpg',
    value: './resources/images/Inception.jpg',
  },
  {
    name: 'Reservoir Dogs.jpg',
    value: './resources/images/Reservoir Dogs.jpg',
  },
];

export default images;
