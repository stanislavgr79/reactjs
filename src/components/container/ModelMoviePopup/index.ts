import ModelMoviePopup from './ModelMoviePopup';

export default ModelMoviePopup;
