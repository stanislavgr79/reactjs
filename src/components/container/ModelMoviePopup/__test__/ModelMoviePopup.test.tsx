import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import ModelMoviePopup from '../ModelMoviePopup';

import { render, screen, fireEvent, act, RenderOptions, waitFor } from '@testing-library/react';
import {
    FETCH_ADD_MOVIE_SUCCESS,
  MoviesActionTypes,
  UPDATE_MOVIE_SUCCESS,
  UPDATE_STATUS_MOVIE_POPUP,
} from '../../../../redux/types/movies-reducer-types';
import { IMovie } from '../../../../helpers/interface';
import userEvent from '@testing-library/user-event';
jest.mock('../../../atom/SiteName', () => {
  return {
    __esModule: true,
    default: () => {
      return <div className="siteName"></div>;
    },
  };
});
jest.mock('../../../structure/Footer', () => {
  return {
    __esModule: true,
    default: () => {
      return <div className="footer"></div>;
    },
  };
});
jest.mock('../../../../redux/actions/movies-actions', () => ({
  fetchUpdateMovie: (movie: IMovie): MoviesActionTypes => {
    return {
      type: UPDATE_MOVIE_SUCCESS,
      payload: movie,
    };
  },
  fetchAddMovie: (movie: IMovie): MoviesActionTypes => {
    return {
      type: FETCH_ADD_MOVIE_SUCCESS,
      payload: movie,
    };
  },
  updateShowPopup: (): MoviesActionTypes => {
    return {
      type: UPDATE_STATUS_MOVIE_POPUP,
      payload: false,
    };
  },
}));

describe('ModelMoviePopup', () => {
  const initialState = { moviesStore: { showPopup: true } };
  const mockStore = configureStore();
  const closePopup = jest.fn();
  let container: Element 
    | DocumentFragment 
    // eslint-disable-next-line prettier/prettier
    | RenderOptions<typeof import('@testing-library/dom/types/queries'), HTMLElement>
    | null;
  const movie = {
    id: 12345,
    title: 'film title',
    release_date: '2020-01-02',
    poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
    genres: ['crime'],
    overview: 'overview',
    runtime: 66,
  };
  
  afterEach(() => {
    container = null;
    jest.clearAllMocks();
  });

  test('render edit model', async () => {
    container = render(<div></div>);
    const store = mockStore(initialState);

    container = render(<Provider store={store}>
        <ModelMoviePopup movie={movie} closePopup={closePopup} role="edit" />
    </Provider>,);
    
    expect(container.baseElement).toMatchSnapshot();
    expect(screen.getByRole('button', { name: /SAVE/i })).toBeInTheDocument();
  });

  test('render add model', async () => {
    container = render(<div></div>);
    const store = mockStore(initialState);

    container = render(
        <Provider store={store}>
            <ModelMoviePopup closePopup={closePopup} role="add" />
        </Provider>,);
    
    expect(container.baseElement).toMatchSnapshot();
    expect(screen.getByRole('button', { name: /SUBMIT/i })).toBeInTheDocument();
  });

  test('closeEditPopup function', () => {
    const store = mockStore(initialState);
    render(
        <Provider store={store}>
            <ModelMoviePopup closePopup={closePopup} role="add" />
        </Provider>,);

    const expectedActions = [{ type: 'UPDATE_STATUS_MOVIE_POPUP', payload: false }];

    fireEvent.click(screen.getByRole('button', { name: '' }));
    expect(closePopup).toBeCalled();
    expect(store.getActions()).toEqual(expectedActions);
  });

  test('handleSubmitEdit function Out description', async () => {
    const promise = Promise.resolve();
    container = render(<div></div>);
    const store = mockStore(initialState);
    const movieExpected = {
      id: 12345,
      title: 'film title',
      release_date: '2020-01-02',
      poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
      genres: ['crime'],
      overview: 'overview',
      runtime: 66,
      tagline: 'Out description',
    };

    act(() => {
        container = render(<Provider store={store}>
            <ModelMoviePopup movie={movie} closePopup={closePopup} role="edit" />
        </Provider>,);
    });

    act(() => {
        fireEvent.click(screen.getByRole('button', { name: 'SAVE' }));
    });

    const expectedActions = [
      { type: 'UPDATE_MOVIE_SUCCESS', payload: movieExpected },
      { type: 'UPDATE_STATUS_MOVIE_POPUP', payload: false },
    ];
    
    await act(() => promise);

    await waitFor(() => {
        expect(store.getActions()).toEqual(expectedActions);
        expect(closePopup).toBeCalled();
    });
  });

  test('handleSubmitEdit function Test description', async () => {
    const promise = Promise.resolve();
    container = render(<div></div>);
    const store = mockStore(initialState);
    const movie = {
      id: 12345,
      title: 'film title',
      release_date: '2020-01-02',
      poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
      genres: ['crime'],
      overview: 'overview',
      runtime: 66,
      tagline: 'Test description',
    };

    act(() => {
        container = render(<Provider store={store}>
            <ModelMoviePopup movie={movie} closePopup={closePopup} role="edit" />
        </Provider>,);
    });

    act(() => {
        fireEvent.click(screen.getByRole('button', { name: 'SAVE' }));
    });

    const expectedActions = [
      { type: 'UPDATE_MOVIE_SUCCESS', payload: movie },
      { type: 'UPDATE_STATUS_MOVIE_POPUP', payload: false },
    ];
    
    await act(() => promise);

    await waitFor(() => {
        expect(store.getActions()).toEqual(expectedActions);
        expect(closePopup).toBeCalled();
    });
  });

  test('handleSubmitAdd function Out description', async () => {
    const promise = Promise.resolve();
    container = render(<div></div>);
    const store = mockStore(initialState);
    const movieSubmit = {
        title: 'film title',
        release_date: "2021-04-30",
        poster_path: 'https://image.tmdb.org/3kcEGBH.jpg',
        genres: ['Action'],
        overview: 'overview',
        runtime: 77,
        budget: 0,
        revenue: 0,
        tagline: 'Out description',
        vote_average: 0,
        vote_count: 0,
      };

    act(() => {
        container = render(<Provider store={store}>
            <ModelMoviePopup closePopup={closePopup} role="add" />
        </Provider>,);
    });

    userEvent.type(screen.getByRole('textbox', { name: 'TITLE' }), "film title");
    userEvent.type(screen.getByRole('textbox', { name: 'OVERVIEW' }), "overview");
    userEvent.type(screen.getByRole('spinbutton', { name: 'RUNTIME' }), '77');
    userEvent.type(screen.getByRole('textbox', { name: 'POSTER URL' }),
            "https://image.tmdb.org/3kcEGBH.jpg");
    userEvent.click(screen.getByRole('img'));
    fireEvent.click(screen.getByRole('button', { name: 'Choose Friday, April 30th, 2021' }));
    userEvent.click(screen.getByRole('textbox', { name: 'GENRE Select Genre' }));
    userEvent.click(screen.getByText('Action'));

    act(() => {
        fireEvent.click(screen.getByRole('button', { name: 'SUBMIT' }));
    });
    
    const expectedActions = [
      { type: 'FETCH_ADD_MOVIE_SUCCESS', payload: movieSubmit },
    ];   
    await act(() => promise);
    expect(container.baseElement).toMatchSnapshot();
    await waitFor(() => {
        expect(store.getActions()).toEqual(expectedActions);
        expect(closePopup).toBeCalled();
    });
  });

  test('reset button function', async () => {
    const promise = Promise.resolve();
    container = render(<div></div>);
    const store = mockStore(initialState);

    act(() => {
        container = render(<Provider store={store}>
            <ModelMoviePopup closePopup={closePopup} role="add" />
        </Provider>,);
    });

    userEvent.type(screen.getByRole('textbox', { name: 'TITLE' }), "film title");
    userEvent.type(screen.getByRole('textbox', { name: 'OVERVIEW' }), "overview");
    userEvent.type(screen.getByRole('spinbutton', { name: 'RUNTIME' }), '77');
    userEvent.type(screen.getByRole('textbox', { name: 'POSTER URL' }),
            "https://image.tmdb.org/3kcEGBH.jpg");
    userEvent.click(screen.getByRole('img'));
    fireEvent.click(screen.getByRole('button', { name: 'Choose Friday, April 30th, 2021' }));
    userEvent.click(screen.getByRole('textbox', { name: 'GENRE Select Genre' }));
    userEvent.click(screen.getByText('Action'));

    act(() => {
        fireEvent.click(screen.getByRole('button', { name: 'RESET' }));
    });
    
    await act(() => promise);
    expect(container.baseElement).toMatchSnapshot();
  });
});
