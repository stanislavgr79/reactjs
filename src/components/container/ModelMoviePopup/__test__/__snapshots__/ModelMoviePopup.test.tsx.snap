// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelMoviePopup handleSubmitAdd function Out description 1`] = `
<body>
  <div>
    <div />
  </div>
  <div>
    <div
      class="popup_model"
    >
      <div
        class="model_content container-md"
      >
        <div
          class="siteName"
        />
        <div
          class="container add_movie"
        >
          <div
            class="container-sm form"
          >
            <div
              class="page_close"
            >
              <button
                class="btn button btn_close"
                id=""
                type="button"
              />
            </div>
            <div
              class="page_title"
            >
              ADD MOVIE
            </div>
            <form
              class="movie_form"
            >
              <label>
                TITLE
                <input
                  class="input simple_input"
                  name="title"
                  placeholder="Write Title"
                  type="text"
                  value="film title"
                />
              </label>
              <label>
                RELEASE DATE
                <div
                  class="react-datepicker-wrapper"
                >
                  <div
                    class="react-datepicker__input-container"
                  >
                    <input
                      class="simple_input"
                      name="release_date"
                      placeholder="Select Date"
                      readonly=""
                      value="2021-04-30"
                    />
                    <img
                      alt=""
                      class="image_calendar"
                      src="[object Object]"
                    />
                  </div>
                </div>
              </label>
              <label>
                POSTER URL
                <input
                  class="input simple_input"
                  name="poster_path"
                  placeholder="Poster URL here"
                  type="text"
                  value="https://image.tmdb.org/3kcEGBH.jpg"
                />
              </label>
              <label>
                GENRE
                <div
                  class="select_genres css-2b097c-container"
                >
                  <div
                    class=" css-yk16xz-control"
                  >
                    <div
                      class=" css-g1d714-ValueContainer"
                    >
                      <div
                        class="css-1rhbuit-multiValue"
                      >
                        <div
                          class="css-12jo7m5"
                        >
                          <span>
                            Action
                          </span>
                        </div>
                        <div
                          class="css-xb97g8"
                        >
                          <svg
                            aria-hidden="true"
                            class="css-tj5bde-Svg"
                            focusable="false"
                            height="14"
                            viewBox="0 0 20 20"
                            width="14"
                          >
                            <path
                              d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        class="css-b8ldur-Input"
                      >
                        <div
                          class=""
                          style="display: inline-block;"
                        >
                          <input
                            aria-autocomplete="list"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            id="react-select-7-input"
                            spellcheck="false"
                            style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                          <div
                            style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      class=" css-1hb7zxy-IndicatorsContainer"
                    >
                      <div
                        aria-hidden="true"
                        class=" css-tlfecz-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                          />
                        </svg>
                      </div>
                      <span
                        class=" css-1okebmr-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class=" css-tlfecz-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div>
                    <input
                      name="genres"
                      type="hidden"
                      value="Action"
                    />
                  </div>
                </div>
              </label>
              <div>
                Must be one or more genres
              </div>
              <label>
                OVERVIEW
                <input
                  class="input simple_input"
                  name="overview"
                  placeholder="Overview here"
                  type="text"
                  value="overview"
                />
              </label>
              <label>
                RUNTIME
                <input
                  class="input simple_input"
                  min="0"
                  name="runtime"
                  placeholder="Runtime here"
                  type="number"
                  value="77"
                />
              </label>
              <div
                class="button_section"
              >
                <button
                  class="btn button btn_submit"
                  disabled=""
                  id="add_movie_btn_submit"
                  type="submit"
                >
                  SUBMIT
                </button>
                <button
                  class="btn button btn_reset"
                  id="add_movie_btn_reset"
                  type="reset"
                >
                  RESET
                </button>
              </div>
            </form>
          </div>
        </div>
        <div
          class="footer"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`ModelMoviePopup render add model 1`] = `
<body>
  <div>
    <div />
  </div>
  <div>
    <div
      class="popup_model"
    >
      <div
        class="model_content container-md"
      >
        <div
          class="siteName"
        />
        <div
          class="container add_movie"
        >
          <div
            class="container-sm form"
          >
            <div
              class="page_close"
            >
              <button
                class="btn button btn_close"
                id=""
                type="button"
              />
            </div>
            <div
              class="page_title"
            >
              ADD MOVIE
            </div>
            <form
              class="movie_form"
            >
              <label>
                TITLE
                <input
                  class="input simple_input"
                  name="title"
                  placeholder="Write Title"
                  type="text"
                  value=""
                />
              </label>
              <label>
                RELEASE DATE
                <div
                  class="react-datepicker-wrapper"
                >
                  <div
                    class="react-datepicker__input-container"
                  >
                    <input
                      class="simple_input"
                      name="release_date"
                      placeholder="Select Date"
                      readonly=""
                      value=""
                    />
                    <img
                      alt=""
                      class="image_calendar"
                      src="[object Object]"
                    />
                  </div>
                </div>
              </label>
              <label>
                POSTER URL
                <input
                  class="input simple_input"
                  name="poster_path"
                  placeholder="Poster URL here"
                  type="text"
                  value=""
                />
              </label>
              <label>
                GENRE
                <div
                  class="select_genres css-2b097c-container"
                >
                  <div
                    class=" css-yk16xz-control"
                  >
                    <div
                      class=" css-g1d714-ValueContainer"
                    >
                      <div
                        class=" css-1wa3eu0-placeholder"
                      >
                        Select Genre
                      </div>
                      <div
                        class="css-b8ldur-Input"
                      >
                        <div
                          class=""
                          style="display: inline-block;"
                        >
                          <input
                            aria-autocomplete="list"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            id="react-select-3-input"
                            spellcheck="false"
                            style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                          <div
                            style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      class=" css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        class=" css-1okebmr-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class=" css-tlfecz-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div>
                    <input
                      name="genres"
                      type="hidden"
                    />
                  </div>
                </div>
              </label>
              <label>
                OVERVIEW
                <input
                  class="input simple_input"
                  name="overview"
                  placeholder="Overview here"
                  type="text"
                  value=""
                />
              </label>
              <label>
                RUNTIME
                <input
                  class="input simple_input"
                  min="0"
                  name="runtime"
                  placeholder="Runtime here"
                  type="number"
                  value=""
                />
              </label>
              <div
                class="button_section"
              >
                <button
                  class="btn button btn_submit"
                  id="add_movie_btn_submit"
                  type="submit"
                >
                  SUBMIT
                </button>
                <button
                  class="btn button btn_reset"
                  id="add_movie_btn_reset"
                  type="reset"
                >
                  RESET
                </button>
              </div>
            </form>
          </div>
        </div>
        <div
          class="footer"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`ModelMoviePopup render edit model 1`] = `
<body>
  <div>
    <div />
  </div>
  <div>
    <div
      class="popup_model"
    >
      <div
        class="model_content container-md"
      >
        <div
          class="siteName"
        />
        <div
          class="container edit_movie"
        >
          <div
            class="container-sm form"
          >
            <div
              class="page_close"
            >
              <button
                class="btn button btn_close"
                id=""
                type="button"
              />
            </div>
            <div
              class="page_title"
            >
              EDIT MOVIE
            </div>
            <form
              class="movie_form"
            >
              <label>
                MOVIE ID
              </label>
              <input
                class="input simple_input input_readonly"
                name="id"
                readonly=""
                type="text"
                value="12345"
              />
              <label>
                TITLE
                <input
                  class="input simple_input"
                  name="title"
                  placeholder="Write Title"
                  type="text"
                  value="film title"
                />
              </label>
              <label>
                RELEASE DATE
                <div
                  class="react-datepicker-wrapper"
                >
                  <div
                    class="react-datepicker__input-container"
                  >
                    <input
                      class="simple_input"
                      name="release_date"
                      placeholder="Select Date"
                      readonly=""
                      value="2020-01-02"
                    />
                    <img
                      alt=""
                      class="image_calendar"
                      src="[object Object]"
                    />
                  </div>
                </div>
              </label>
              <label>
                POSTER URL
                <input
                  class="input simple_input"
                  name="poster_path"
                  placeholder="Poster URL here"
                  type="text"
                  value="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
                />
              </label>
              <label>
                GENRE
                <div
                  class="select_genres css-2b097c-container"
                >
                  <div
                    class=" css-yk16xz-control"
                  >
                    <div
                      class=" css-g1d714-ValueContainer"
                    >
                      <div
                        class="css-1rhbuit-multiValue"
                      >
                        <div
                          class="css-12jo7m5"
                        >
                          <span>
                            crime
                          </span>
                        </div>
                        <div
                          class="css-xb97g8"
                        >
                          <svg
                            aria-hidden="true"
                            class="css-tj5bde-Svg"
                            focusable="false"
                            height="14"
                            viewBox="0 0 20 20"
                            width="14"
                          >
                            <path
                              d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        class="css-b8ldur-Input"
                      >
                        <div
                          class=""
                          style="display: inline-block;"
                        >
                          <input
                            aria-autocomplete="list"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            id="react-select-2-input"
                            spellcheck="false"
                            style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                          <div
                            style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      class=" css-1hb7zxy-IndicatorsContainer"
                    >
                      <div
                        aria-hidden="true"
                        class=" css-tlfecz-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                          />
                        </svg>
                      </div>
                      <span
                        class=" css-1okebmr-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class=" css-tlfecz-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div>
                    <input
                      name="genres"
                      type="hidden"
                      value="crime"
                    />
                  </div>
                </div>
              </label>
              <label>
                OVERVIEW
                <input
                  class="input simple_input"
                  name="overview"
                  placeholder="Overview here"
                  type="text"
                  value="overview"
                />
              </label>
              <label>
                RUNTIME
                <input
                  class="input simple_input"
                  min="0"
                  name="runtime"
                  placeholder="Runtime here"
                  type="number"
                  value="66"
                />
              </label>
              <div
                class="button_section"
              >
                <button
                  class="btn button btn_submit"
                  id="add_movie_btn_submit"
                  type="submit"
                >
                  SAVE
                </button>
                <button
                  class="btn button btn_reset"
                  id="add_movie_btn_reset"
                  type="reset"
                >
                  RESET
                </button>
              </div>
            </form>
          </div>
        </div>
        <div
          class="footer"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`ModelMoviePopup reset button function 1`] = `
<body>
  <div>
    <div />
  </div>
  <div>
    <div
      class="popup_model"
    >
      <div
        class="model_content container-md"
      >
        <div
          class="siteName"
        />
        <div
          class="container add_movie"
        >
          <div
            class="container-sm form"
          >
            <div
              class="page_close"
            >
              <button
                class="btn button btn_close"
                id=""
                type="button"
              />
            </div>
            <div
              class="page_title"
            >
              ADD MOVIE
            </div>
            <form
              class="movie_form"
            >
              <label>
                TITLE
                <input
                  class="input simple_input"
                  name="title"
                  placeholder="Write Title"
                  type="text"
                  value=""
                />
              </label>
              <label>
                RELEASE DATE
                <div
                  class="react-datepicker-wrapper"
                >
                  <div
                    class="react-datepicker__input-container"
                  >
                    <input
                      class="simple_input"
                      name="release_date"
                      placeholder="Select Date"
                      readonly=""
                      value=""
                    />
                    <img
                      alt=""
                      class="image_calendar"
                      src="[object Object]"
                    />
                  </div>
                </div>
              </label>
              <label>
                POSTER URL
                <input
                  class="input simple_input"
                  name="poster_path"
                  placeholder="Poster URL here"
                  type="text"
                  value=""
                />
              </label>
              <label>
                GENRE
                <div
                  class="select_genres css-2b097c-container"
                >
                  <div
                    class=" css-yk16xz-control"
                  >
                    <div
                      class=" css-g1d714-ValueContainer"
                    >
                      <div
                        class=" css-1wa3eu0-placeholder"
                      >
                        Select Genre
                      </div>
                      <div
                        class="css-b8ldur-Input"
                      >
                        <div
                          class=""
                          style="display: inline-block;"
                        >
                          <input
                            aria-autocomplete="list"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            id="react-select-8-input"
                            spellcheck="false"
                            style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                          <div
                            style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      class=" css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        class=" css-1okebmr-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class=" css-tlfecz-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div>
                    <input
                      name="genres"
                      type="hidden"
                    />
                  </div>
                </div>
              </label>
              <label>
                OVERVIEW
                <input
                  class="input simple_input"
                  name="overview"
                  placeholder="Overview here"
                  type="text"
                  value=""
                />
              </label>
              <label>
                RUNTIME
                <input
                  class="input simple_input"
                  min="0"
                  name="runtime"
                  placeholder="Runtime here"
                  type="number"
                  value=""
                />
              </label>
              <div
                class="button_section"
              >
                <button
                  class="btn button btn_submit"
                  id="add_movie_btn_submit"
                  type="submit"
                >
                  SUBMIT
                </button>
                <button
                  class="btn button btn_reset"
                  id="add_movie_btn_reset"
                  type="reset"
                >
                  RESET
                </button>
              </div>
            </form>
          </div>
        </div>
        <div
          class="footer"
        />
      </div>
    </div>
  </div>
</body>
`;
